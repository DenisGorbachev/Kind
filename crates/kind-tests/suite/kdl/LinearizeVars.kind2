U120 : Type
U120.new (a: U60) (b: U60) : U120

#kdl_erase
U120.add (a: U120) (b: U120) : U120

List : Type
List.nil : List
List.cons (h: U120) (t: List) : Type

#keep
TestFunc (xs: List) : List
TestFunc (List.cons h t) =
    let aa = 2u120
    let bb = (U120.add aa aa)
    let cc = 4u120
    List.cons (U120.add aa bb) t