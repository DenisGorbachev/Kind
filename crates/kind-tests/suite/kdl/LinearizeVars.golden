ctr {List.cons h t}

fun (U120.new hi lo) {
  (U120.new x0 x1) = (!@x1.0 (!@x0.0 (| (<< x0.0 #60) x1.0) x0) x1)
}

fun (TestFunc xs) {
  (TestFunc {List.cons ~ x1}) = (!@x1.0 (!@x2 dup c.0 x2.0 = x2; dup x2.1 x2.2 = c.0; (!@x3 (!@x3.0 (!@~ {List.cons (& (+ x2.0 x3.0) #1152921504606846975) x1.0} #4) x3) (& (+ x2.1 x2.2) #1152921504606846975)) #2) x1)
}

